portfolio/src/main/webapp/pagination.js

```diff
@@ -0,0 +1,185 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+document.addEventListener('DOMContentLoaded', () => {
+  const commentButton = new CommentButton();
+  commentButton.requestAndDisplayNewCommentsPage();
+});
+
+class CommentButton {
+  constructor() {
+    this.currentCommentsPageIndex = 0;
+    this.totalNumberOfPagesRequested = 1;
+    this.lastPage = -1;
+    this.cursorString = null;
+    this.prevCommentsButton = document.getElementById('prev-comments-button');
+    this.nextCommentsButton = document.getElementById('next-comments-button');
+    this.endOfCommentsSign = document.getElementById('end-of-comments-sign');
+    this.comments = document.getElementById('comments');
+
+    this.prevCommentsButton.classList.add("hidden");
+    this.endOfCommentsSign.classList.add("hidden");
+   
+    document.getElementById('maxNum').addEventListener('input', () => {
+      // Resetting innerHTML of comments when user re-selects the number of maximum comments to display
+      this.comments.innerHTML = "";
+      // Displays most recent page of comments so "prev" button should be hidden, "end-of-comments-sign" should be hidden, while "next" button should display.
+      this.prevCommentsButton.classList.add("hidden");
+      this.nextCommentsButton.classList.remove("hidden");
+      this.endOfCommentsSign.classList.add("hidden");
+
+      this.currentCommentsPageIndex = 0;
+      this.totalNumberOfPagesRequested = 1;
+      this.lastPage = -1;
+      this.requestAndDisplayNewCommentsPage();
+    });
+
+    this.prevCommentsButton.addEventListener('click', () => {
+      this.displayPrevPage();
+    });
+
+    this.nextCommentsButton.addEventListener('click', () => {
+      this.displayNextPage();
+    });
+  }
+
+  hideCommentAtIndex(index) {
+    const commentList = this.comments.children;
+    commentList[index].classList.add("hidden");
+  }
+
+  showCommentAtIndex(index) {
+    const commentList = this.comments.children;
+    commentList[index].classList.remove("hidden");
+  }
+
+  disableButtons() {
+    this.prevCommentsButton.disabled = true;
+    this.nextCommentsButton.disabled = true;
+  }
+
+  enableButtons() {
+    this.prevCommentsButton.disabled = false;
+    this.nextCommentsButton.disabled = false;
+  }
+
+   async displayNextPage() {
+    this.disableButtons();
+
+    this.prevCommentsButton.classList.remove("hidden");
+
+    // this.currentCommentsPageIndex points to the page that is displayed before the user clicks next
+    // if condition checks whether the page to display when user clicks next is the last page, which holds -1 or the last page index that is set in the request/display new comments page function
```

Comment:
this is a little too verbose; instead put the comment in the if and make it shorter;

// There are no more comments.